src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1308,6 +1313,20 @@ internal NamedTypeSymbol GetFixedImplementationType(FieldSymbol field)
             return result;
         }
 
+        internal Cci.IFieldReference GetModuleVersionId(Cci.ITypeReference mvidType, CSharpSyntaxNode syntaxOpt, DiagnosticBag diagnostics)
+        {
+            PrivateImplementationDetails details = GetPrivateImplClass(syntaxOpt, diagnostics);
+            // Avoid creating a closure for the static constructor creator unless it is likely to be executed.
+            return details.HasStaticConstructor ?
+                details.GetOrAddModuleVersionId(mvidType, null) :
+                GetModuleVersionIdWithStaticConstructorCreation(mvidType, syntaxOpt, diagnostics);
+        }
+
+        private Cci.IFieldReference GetModuleVersionIdWithStaticConstructorCreation(Cci.ITypeReference mvidType, CSharpSyntaxNode syntaxOpt, DiagnosticBag diagnostics) =>
+            PrivateImplClass.GetOrAddModuleVersionId(
+                mvidType,
+                () => new SynthesizedPrivateImplementationDetailsStaticConstructor(SourceModule, PrivateImplClass, GetUntranslatedSpecialType(SpecialType.System_Void, syntaxOpt, diagnostics)));
```

Comment:
OK. This does make it theoretically possible to create a field and not create a constructor to initialize it, but that's a very low risk.

My thinking was to make this impossible by requiring a static constructor creator to be supplied in all circumstances (including future ones) that required a constructor. However, as suggested in a later review comment, the code is less tangled if we avoid delegates altogether.